velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -778,25 +728,40 @@ std::shared_ptr<exec::VectorFunction> createDecimalFunction(
     }
   }
 }
+
+// Find the function according to input types under expression compiler.
+// Mark the return type as unknown, the result type will be computed by Function
```

Comment:
This is a hack. Let's not do this.

It seems like function signatures are affected by configuration settings. If that's the case, let's come up with a way to allow access to configs during function registration time.

We do need to be able to resolve function calls without instantiating functions. This is what enables Fuzzer and unit testing.